### PR TITLE
feat: [Subscales] Add Severity to Lookup Table (CAMHI) (M2-7586)

### PR DESCRIPTION
**Pull request by sultanofcardio**
- [x] Tests for the changes have been added

### 📝 Description

🔗 [Jira Ticket M2-7586](https://mindlogger.atlassian.net/browse/M2-7586)

> [!NOTE]
> This feature is behind the flag `enableCahmiSubscaleScoring`

This PR adds support for the severity column in the subscales lookup table. This includes the following changes:
- The subscales lookup table template now includes a severity column
- The parsing code has been updated to support the proper severity values (Minimal, Mild, Moderate, Severe) and the empty string
- The age column in the lookup table now supports ranges using the interval symbol ~ (e.g. 10~15). This is similar to the ranges supported by the `score` and `rawScore` columns
- The lookup table preview has been updated to show the severity column
- The severity values are being saved to the database

#### CSV Template

image image

#### Lookup Table Preview

image

image

### 🪤 Peer Testing

1. Edit an existing activity
2. Add an item with a score attached (such as a Single Selection item)
3. Go to the Activity Settings tab
4. Click Subscales Configuration
5. Create a subscale
6. Click the little image icon
7. Download the template
8. Re-upload the template without editing
9. Confirm that the preview includes the severity column, and a warning is shown for the incomplete severity column
10. Click save on the modal
11. Click the Save & Publish button
12. Check the database and confirm the presence of the `severity` property in the `subscale_setting` JSONB column of the `activities` table
    image
13. Repeat the test with invalid severity values and confirm that parsing fails
14. Turn off the feature flag and confirm that the current behaviour functions as expected

### ✏️ Notes

1. ~Helper text that explains the various severity levels is coming in the follow-up ticket M2&#8722;7747~

### CODE REVIEW

File: src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/LookupTable/LookupTable.hooks.ts [hunk 1]
```diff
@@ -29,7 +40,29 @@ export const useSubscaleLookupTableSetup = ({
...
+        mappedItem.severity = (item.severity as TScoreSeverity) || null;
+      }
+
+      return mappedItem;
+    });
+
+    const incompleteSeverityData = mappedData.some((item) => !item.severity);
+    if (incompleteSeverityData) {
+      setWarning(warnings.incompleteSeverityData);
+    }
```
Comment (farmerpaul) [inline src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/LookupTable/LookupTable.hooks.ts]: Should these lines also be behind the feature flag? I'm getting this when it's turned off, and I upload a CSV that lacks the `severity` column:
CleanShot 2024-09-16 at 17 32 15@2x

**Review by farmerpaul — APPROVED** [review]
Just discovered a minor bug with the feature flag turned off, but seems like a simple fix, so preapproving. All other AC behave as expected. :+1:

File: src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/LookupTable/LookupTable.hooks.ts
Comment (mbanting) [inline src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/LookupTable/LookupTable.hooks.ts:1]: Is this where tests will be added for complete and incomplete severity data?

**Review by mbanting — COMMENTED** [review]
Great job @sultanofcardio . 
Did some quick testing and it works well. Just the feature flag on the severity check, as @farmerpaul pointed out. 

Noticed in the description that tests are still in progress. I can have a look again once those are added but PR is looking great so far.

File: src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/LookupTable/LookupTable.hooks.ts (hunk 1 above)
Comment (sultanofcardio) [reply src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/LookupTable/LookupTable.hooks.ts]: Thanks for catching this Paul 🙏

File: src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/LookupTable/LookupTable.hooks.ts
Comment (sultanofcardio) [reply src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/LookupTable/LookupTable.hooks.ts:1]: Thanks for reviewing Marty. I've been adding my tests to this file https://github.com/ChildMindInstitute/mindlogger-admin/blob/e45814f1f8bd90f9f34d5df75403f138221ae56a/src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscalesConfiguration.test.tsx#L185

Comment (sultanofcardio) [reply src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/LookupTable/LookupTable.hooks.ts:1]: @Marty tests have now been added

**Comment by sultanofcardio** [comment]
@mbanting and @farmerpaul I've added some tests, and updated the code to address the new AC for supporting age ranges. Please give it a re-review when you have a minute 🙏

File: src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/LookupTable/LookupTable.hooks.ts
Comment (mbanting) [reply src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/LookupTable/LookupTable.hooks.ts:1]: Thanks!

**Review by mbanting — APPROVED** [review]
Looks great @sultanofcardio . Good set of tests. 
Noticed that it's possible for the age range to be 10~5 (ie. first number large than second number). It wasn't called out in the AC so happy to approve here but raised it in a [comment](https://mindlogger.atlassian.net/browse/M2-7586?focusedCommentId=24911) on the ticket, suggesting we tackle this in a followup.

**Review by farmerpaul — APPROVED** [review]

**Pull request merged**